src/MAGMA.jl

```diff
@@ -23,15 +23,140 @@ const MagmaSomeVec       = 305
 const MagmaOverwriteVec  = 306
 const MagmaBacktransVec  = 307
 
+# https://github.com/mweastwood/MAGMA.jl
+
+export device_reset,
+       MagmaVector, MagmaMatrix, get,
+       axpy!, dot,
+       gemv!
+
+import Base: length,size,dot
+
+
+# some types' definitions and their accordingly chararcters
+const magmatypes = Union{Float32, Float64, ComplexF32, ComplexF64}
+const magmatypeslist =  (Float32,Float64,ComplexF32,ComplexF64)
+const typechar = Dict(
+    Float32 => 's',
+    Float64 => 'd',
+    ComplexF32 => 'c',
+    ComplexF64 => 'z'
+)
+
 """
 the path to magma binary
 """
 const libmagma = "/usr/local/magma/lib/libmagma.so"
 
+
+# https://github.com/mweastwood/MAGMA.jl
+struct MagmaVector{T <: magmatypes}
```

Comment:
Create a `MagmaArray` to be subtype of `AbstractArray` and make use of existing Array interface

Reply:
A question: Why you can't use `unsafe_wrap` and `Base.unsafe_convert` to wrap the pointer to `Array` on CPU but instead define your own type?

Reply:
I tried the default convert function in Julia stdlib and found that it indeed works... So now I directly convert them without call magma's utility functions...
I guess maybe this will bring some loss in efficiency? Anyway, I suggest that we put it off currently and come back to its improvement after finishing all the binding.